pkg/kubectl/resource_printer.go

```diff
@@ -1174,3 +1184,26 @@ func indirect(v reflect.Value) (rv reflect.Value, isNil bool) {
 	}
 	return v, false
 }
+
+// JSONPathPrinter is an implementation of ResourcePrinter which formats data with jsonpath expression.
+type JSONPathPrinter struct {
+	rawTemplate string
+	template    *jsonpath.JSONPath
```

Comment:
if you remove the rawTemplate member, then I would just make this an embedded field, instead of a named field.

https://golang.org/doc/effective_go.html#embedding